src/usb-export

```diff
@@ -77,63 +80,70 @@ if [ -z "$devpath" ]; then
     exit 1
 fi
 
-pidfile="/var/run/qubes/usb-export-$(basename $devpath).pid"
+busid=${devpath##*/}
+if ! [[ "$busid" =~ ^[0-9]+-[0-9]+$ ]]; then
+    printf 'Invalid bus path %q\n' "$devpath" >&2
+    exit 1
+fi
+pidfile="/var/run/qubes/usb-export-$busid.pid"
 
 modprobe usbip-host || exit 1
 
 # Request that both IN and OUT be handled on a single (stdin) socket
 kill -USR1 "$QREXEC_AGENT_PID" || exit 1
 
-busid=$(basename "$devpath")
 # Unbind the device from the driver
 if [ -d "$devpath/driver" ]; then
-    echo $busid > $devpath/driver/unbind || exit 1
+    printf %s "$busid" > "$devpath/driver/unbind" || exit 1
 fi
 
 # Bind to the usbip-host driver
-echo -n add $busid > $SYS_USBIP_HOST/match_busid || exit 1
-echo $busid > $SYS_USBIP_HOST/bind || exit 1
+printf 'add %s' "$busid" > "$SYS_USBIP_HOST/match_busid" || exit 1
+echo "$busid" > "$SYS_USBIP_HOST/bind" || exit 1
 
 # One more safety check - make sure the device is available
-if [ "$(cat $devpath/usbip_status)" -ne $SDEV_ST_AVAILABLE ]; then
-    echo "Device $devpath not available!" >&2
+read status < "$devpath/usbip_status"
+if [ "$status" -ne "$SDEV_ST_AVAILABLE" ]; then
+    printf 'Device %s not available!\n' "$devpath" >&2
     exit 1
 fi
 
-busnum=$(cat $devpath/busnum)
-devnum=$(cat $devpath/devnum)
-devid=$(( $busnum << 16 | $devnum ))
-speed=$(cat $devpath/speed)
+# Allow the device.
+if command -v usbguard > /dev/null; then
+    usbguard allow-device "via-port \"$busid\""
```

Comment:
Just having usbguard installed doesn't mean it's also running. And with `set -e`, it would fail attaching.

Reply:
Should the service be started in this case?

Reply:
Definitely not. If user didn't enabled it (using whatever method), just using qvm-usb should not start it either.